### ISSUE TITLE
Great project! Mind connecting it with containerized Virtualization?

### ISSUE THREAD

**Issue by boredsquirrel** [hardcore-sushi/hidemypussy#2]
[Luca, the creator of Distrobox, has created a great guide on how to use libvirt, qemu etc. in a Distrobox Container and then connect that to your client in another container using ssh](https://github.com/89luca89/distrobox/blob/main/docs/posts/run_libvirt_in_distrobox.md)

As far as I understood this has many benefits. If the containers are adapted to have the right filesystem permissions and all, they should be pretty tight. I imagine only /var/lib/libvirt and ssh, thats it. The client, for example virt-manager could be on your host OS or even also in a container. I tried this and it works great!

And the tor bridge would sit in another container, tight apart from the tor traffic, but no filesystem permissions at all.

The current guide is not hardened in that way, its simply "make this run on an immutable System without layering anything". And its great! But I think this also has greeeat potential for a really secure Qubes-like OS, but not using virtualization like crazy, but well isolated containers.

What do you think?

**Comment by hardcore-sushi** [comment]
As far as I understand, the only purpose of this guide is to use Libvirt on immutable systems. However, it seems that the Libvirt container needs some privileges to work that could be used to escape to the host. So I don't think it adds any security benefit to put Libvirt in a container.

However, the client can be run inside a container or a sandbox, but this is out of scope for this projoct since TorVirt does not deal with clients.

**Comment by boredsquirrel** [comment]
following the guide exactly this could be true.

But what if you would create a user and put it in the libvirt group? This should make isolation possible,

**Comment by hardcore-sushi** [comment]
Yes you can run the client without root privileges if the user is in the `libvirt` group.

TorVirt uses by default `qemu:///system`, but you can change this by editing the `LIBVIRT_DEFAULT_URI` variable in the `torvirt` script.
